apps/predictions/lib/predictions_pub_sub.ex

```diff
@@ -1,20 +1,24 @@
 defmodule Predictions.PredictionsPubSub do
   @moduledoc """
-  Allow channels to subscribe to prediction streams.
+  Allow channels to subscribe to prediction streams, which are collected into a
+  Map with keys representing route-stop pairs as "<route-id>@<parent-stop-id>"
   """
 
   use GenServer
 
   alias Predictions.{Prediction, StreamSupervisor}
   alias Routes.Route
+  alias Stops.Stop
 
-  defstruct predictions_by_route_id: %{}
+  defstruct predictions_by_key: %{}
 
   @type t :: %__MODULE__{
-          predictions_by_route_id: predictions_by_route_id()
+          predictions_by_key: predictions_by_key()
```

Comment:
suggestion (non-blocking): At some point (especially if we see performance issues with this), it might be worth considering using an `ets` table for this data. It could use `route`, `stop`, and `direction` as keys.

Reply:
We'd have to read/write to the table super frequently no? An interesting thing to keep in mind for sure! Let's see how well this works out without ETS first.

Reply:
reading/writing super-frequently (especially from different processes) is basically the reason you'd use ETS :)